### PR TITLE
Added docs to svg components

### PR DESCRIPTION
**Pull request by rundmt**
Enyo-DCO-1.1-Signed-off-by: Derek Tor derek.tor@lge.com

### CODE REVIEW

File: src/Animate.js [hunk 1]
```diff
@@ -11,16 +16,116 @@ var
...
+    * @type {String}
+    * @default 'd'
+    * @public
+    */
 	attributeName: 'd',
+	/**
+    * A property to determine the repeat count of an animation
+    *
+    * @type {Number}
+    * @default 'd'
```
Comment (webOS101) [inline src/Animate.js]: these defaults seem.... suspicious...

Comment (rundmt) [inline src/Animate.js]: whoops I must have forgot to update these.

**Comment by rundmt** [comment]
updated defaults for SVG.

File: src/Definitions.js [hunk 2]
```diff
@@ -8,9 +12,39 @@ var
...
 module.exports = kind({
+	/**
+	* @private
+	*/
 	name: 'Definitions',
+	/**
+	* @private
+	*/
 	kind: Element,
 	tag: 'defs'
```
Comment (webOS101) [inline src/Definitions.js]: private?

File: src/Element.js [hunk 3]
```diff
@@ -12,6 +16,41 @@ var
...
+*
+* @class Element
+* @extends module:enyo/Control~Control
+* @ui
+* @public
+*/
+
 // Base-kind for deriving all SVG kinds
 module.exports = kind({
 	name: 'Element',
```
Comment (webOS101) [inline src/Element.js]: private?

File: src/Group.js [hunk 4]
```diff
@@ -8,9 +12,39 @@ var
...
 module.exports = kind({
+	/**
+	* @private
+	*/
 	name: 'Group',
+	/**
+	* @private
+	*/
 	kind: Element,
 	tag: 'g',
```
Comment (webOS101) [inline src/Group.js]: private

File: src/Image.js [hunk 5]
```diff
@@ -8,9 +12,38 @@ var
...
 module.exports = kind({
+	/**
+	* @private
+	*/
 	name: 'Image',
+	/**
+	* @private
+	*/
 	kind: Element,
 	tag: 'image',
```
Comment (webOS101) [inline src/Image.js]: private

File: src/Line.js [hunk 6]
```diff
@@ -8,9 +12,35 @@ var
...
 module.exports = kind({
+	/**
+	* @private
+	*/
 	name: 'Line',
+	/**
+	* @private
+	*/
 	kind: Shape,
 	tag: 'line',
```
Comment (webOS101) [inline src/Line.js]: private

File: src/Tspan.js [hunk 7]
```diff
@@ -8,9 +12,47 @@ var
...
+*
+* @class Tspan
+* @extends module:enyo/Text~Text
+* @ui
+* @public
+*/
+
+
+
 // https://developer.mozilla.org/en-US/docs/Web/SVG/Element/tspan
```
Comment (webOS101) [inline src/Tspan.js]: We can probably take this link out and some of the newlines above

**Comment by rundmt** [comment]
Updated animation keyframe default, and added some more docs.

File: src/Animate.js [hunk 8]
```diff
@@ -28,42 +132,95 @@ module.exports = kind({
...
+    * @public
+    */
+	keyframes: null,
+	/**
+    * Keyframe Library is a property that accpepts and object.
+    * This object can contain keys and values with the values
+    * being an array keyframe animations
+    *
+    * @type {Object}
+    * @default null
```
Comment (webOS101) [inline src/Animate.js]: default should stay []

File: src/Animate.js [hunk 9]
```diff
@@ -28,42 +132,95 @@ module.exports = kind({
...
+	*/
 	transforms: {
 		dur: function (val) { return isNaN(val) ? val : val + 'ms'; }
 	},
 	// Methods
+	/**
+	* @private
+	*/
+	constructor: function() {
+        this.keyframes = [];
```
Comment (webOS101) [inline src/Animate.js]: spacing issue here

**Comment by rundmt** [comment]
updated spacing and put back empty array.

File: src/Tspan.js [hunk 10]
```diff
@@ -8,9 +12,44 @@ var
...
+*
+* 		{kind: Root, viewBox: '0 0 100 100', style: 'height: 250px; width: 250px;', components: [
+* 			{kind: SvgText, classes: 'svg-sample-header-text', components: [
+*	 			{kind: Tspan, content: 'hello world', style: 'color: red'}
+* 			]}
+* 		]}
+* ```
+*
+* @class Tspan
+* @extends module:enyo/Text~Text
```
Comment (webOS101) [inline src/Tspan.js]: This should be svg/Text~Text

**Comment by rundmt** [comment]
Updated modules to extend from SVG instead of Enyo

**Pull request merged**